cpp/webcrypto/crypto_ec.cpp

```diff
@@ -331,4 +331,110 @@ jsi::Value GetEcKeyDetail(jsi::Runtime &rt,
+    } else {
+      config.param_encoding = encoding;
+    }
+  } else {
+    throw jsi::JSError(rt, "Invalid param_encoding specified (not int)");
+  }
+
+  // rest of args for encoding
+  unsigned int offset = 3;
+
+  config.public_key_encoding = ManagedEVPPKey::GetPublicKeyEncodingFromJs(
+      rt, args, &offset, kKeyContextGenerate);
+
+  auto private_key_encoding = ManagedEVPPKey::GetPrivateKeyEncodingFromJs(
+      rt, args, &offset, kKeyContextGenerate);
+
+  if (!private_key_encoding.IsEmpty()) {
+    config.private_key_encoding = private_key_encoding.Release();
+  }
+
+  return config;
+}
+
+EVPKeyCtxPointer setup(std::shared_ptr<EcKeyPairGenConfig> config) {
+  EVPKeyCtxPointer key_ctx;
+  switch (config->curve_nid) {
+    case EVP_PKEY_ED25519:
+      // Fall through
+    case EVP_PKEY_ED448:
+      // Fall through
+    case EVP_PKEY_X25519:
+      // Fall through
+    case EVP_PKEY_X448:
+      key_ctx.reset(EVP_PKEY_CTX_new_id(config->curve_nid, nullptr));
+      break;
+    default: {
+      EVPKeyCtxPointer param_ctx(EVP_PKEY_CTX_new_id(EVP_PKEY_EC, nullptr));
+      EVP_PKEY* raw_params = nullptr;
+      if (!param_ctx ||
+          EVP_PKEY_paramgen_init(param_ctx.get()) <= 0 ||
+          EVP_PKEY_CTX_set_ec_paramgen_curve_nid(
+              param_ctx.get(), config->curve_nid) <= 0 ||
+          EVP_PKEY_CTX_set_ec_param_enc(
+              param_ctx.get(), config->param_encoding) <= 0 ||
+          EVP_PKEY_paramgen(param_ctx.get(), &raw_params) <= 0) {
+        return EVPKeyCtxPointer();
+      }
+      EVPKeyPointer key_params(raw_params);
+      key_ctx.reset(EVP_PKEY_CTX_new(key_params.get(), nullptr));
+    }
+  }
+
+  if (key_ctx && EVP_PKEY_keygen_init(key_ctx.get()) <= 0)
+    key_ctx.reset();
+
+  return key_ctx;
+}
+
+std::pair<jsi::Value, jsi::Value> generateEcKeyPair(jsi::Runtime& runtime,
+                                                    std::shared_ptr<EcKeyPairGenConfig> config)
+{
+  // TODO: this is all copied from MGLRsa.cpp - template it up like Node?
+
+  EVPKeyCtxPointer ctx = setup(config);
+
+  if (!ctx) {
+    throw jsi::JSError(runtime, "Error on key generation job");
+  }
+
+  // Generate the key
+  EVP_PKEY* pkey = nullptr;
+  if (!EVP_PKEY_keygen(ctx.get(), &pkey)) {
+    throw jsi::JSError(runtime, "Error generating key");
+  }
+
+  config->key = ManagedEVPPKey(EVPKeyPointer(pkey));
+
+  jsi::Value publicBuffer =
+      ManagedEVPPKey::ToEncodedPublicKey(runtime, std::move(config->key),
+                                         config->public_key_encoding);
+  jsi::Value privateBuffer =
+      ManagedEVPPKey::ToEncodedPrivateKey(runtime, std::move(config->key),
+                                          config->private_key_encoding);
+
+  if (publicBuffer.isUndefined() || privateBuffer.isUndefined()) {
+    throw jsi::JSError(runtime, "Failed to encode public and/or private key (EC)");
+  }
+
+  return {std::move(publicBuffer), std::move(privateBuffer)};
```

Comment:
I prefer explicit initialization instead of C++'s implicit `{` initializer calls - but I guess that's also how it's done in the other file?